ext/ddtrace.c

```diff
@@ -906,6 +897,22 @@ static PHP_RSHUTDOWN_FUNCTION(ddtrace) {
     dd_shutdown_hooks_and_observer();
 
     DDTRACE_G(in_shutdown) = false;
+}
+
+static PHP_RSHUTDOWN_FUNCTION(ddtrace) {
+    UNUSED(module_number, type);
+
+    zend_hash_destroy(&DDTRACE_G(traced_spans));
+
+    if (!get_DD_TRACE_ENABLED()) {
+        if (!DDTRACE_G(disable)) {
```

Comment:
Can you explain why we need both of these checks?

Reply:
Interesting that git marks this as new code :-D In fact these lines were not touched at all.

But if I see that correctly `dd_shutdown_hooks_and_observer()` is called in `dd_force_shutdown_tracing()` (previously bottom of RSHUTDOWN). This function must be called in any case, but it must not be called if the tracer was minit-time disabled.